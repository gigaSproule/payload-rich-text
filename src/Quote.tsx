import type { TextData } from "./Text";
import { Text } from "./Text";

import { Options } from "./types.js";

export interface QuoteData {
  type: "quote";
  children: TextData[];
  direction: "ltr" | "rtl" | null;
  format: string;
  indent: number;
  version: number;
}

export type Props = { data: QuoteData; options?: Options };

export const Quote = ({ data, options }: Props) => {
  if (data.children.length === 0) {
    return null;
  }
  const children = data.children.map((headingChild) => {
    return (
      <Text key={headingChild.text} data={headingChild} options={options} />
    );
  });
  return options?.quote ? (
    options.quote(data, children)
  ) : (
    <blockquote>{children}</blockquote>
  );
};
