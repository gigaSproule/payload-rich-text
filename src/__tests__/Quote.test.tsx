import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../Quote";
import { Quote } from "../Quote";
import { createQuoteData } from "./helpers/createQuoteData";

describe("Quote", () => {
  it("should not render anything if there are no children", () => {
    const quoteData: Props["data"] = createQuoteData({ children: [] });
    const { container } = render(<Quote data={quoteData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render default blockquote if no options provided", () => {
    const quoteData: Props["data"] = createQuoteData();
    const { container } = render(<Quote data={quoteData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render default blockquote if options provided without quote", () => {
    const quoteData: Props["data"] = createQuoteData();
    const { container } = render(<Quote data={quoteData} options={{}} />);
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with quote", () => {
    const quoteData: Props["data"] = createQuoteData();
    const { container } = render(
      <Quote
        data={quoteData}
        options={{
          quote: (quote, children) => {
            return (
              <div>
                <span>Custom Quote - {JSON.stringify(quote)}</span>
                {children}
              </div>
            );
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
