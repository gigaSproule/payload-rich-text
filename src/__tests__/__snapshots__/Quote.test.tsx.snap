// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should not render anything if there are no children 1`] = `<div />`;

exports[`Quote should render custom render if options provided with quote 1`] = `
<div>
  <div>
    <span>
      Custom Quote - 
      {"children":[{"detail":0,"format":0,"mode":"normal","style":"","text":"heading text","type":"text","version":1}],"type":"quote","direction":null,"format":"","indent":0,"version":1}
    </span>
    heading text
  </div>
</div>
`;

exports[`Quote should render default blockquote if no options provided 1`] = `
<div>
  <blockquote>
    heading text
  </blockquote>
</div>
`;

exports[`Quote should render default blockquote if options provided without quote 1`] = `
<div>
  <blockquote>
    heading text
  </blockquote>
</div>
`;
